connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -90,26 +132,48 @@ public void start() throws IOException {
      * Clean up any temp directories created locally.
      */
     public void stop() {
-        for (Connect worker : this.connectCluster) {
-            try {
-                worker.stop();
-            } catch (Exception e) {
-                log.error("Could not stop connect", e);
-                throw new RuntimeException("Could not stop worker", e);
-            }
-        }
-
+        connectCluster.forEach(this::stopWorker);
         try {
             kafkaCluster.after();
         } catch (Exception e) {
             log.error("Could not stop kafka", e);
             throw new RuntimeException("Could not stop brokers", e);
+        } finally {
+            Exit.resetExitProcedure();
+            Exit.resetHaltProcedure();
+        }
+    }
+
+    /**
+     * Provision and start an additional worker to the Connect cluster.
+     */
+    public void addWorker() {
+        connectCluster.add(new ConnectDistributed().startConnect(workerProps));
+    }
+
+    /**
+     * Decommission a worker from this Connect cluster.
+     */
+    public void removeWorker() {
+        if (connectCluster.isEmpty()) {
+            throw new IllegalStateException("Cannot remove worker. Cluster is empty");
+        }
+        Connect worker = connectCluster.remove(connectCluster.size() - 1);
+        stopWorker(worker);
+    }
+
+    private void stopWorker(Connect worker) {
+        try {
+            worker.stop();
```

Comment:
Might be nice to have log messages (at least at DEBUG) stating that the specific connector worker is being stopped.

Reply:
Interestingly, the logger for integration tests is `OFF`. I'm manually turning it one with a temporary commit so far. Before I suggest a broader change, I'd like to think a bit more for which packages we should turn logging on if we move this way. 

However, I'll add the equivalent log message that we have in `startConnect` to preserve symmetry